staging/src/slime.io/slime/modules/limiter/controllers/producer.go

```diff
@@ -72,45 +71,76 @@ func (r *SmartLimiterReconciler) handleTickerEvent(event trigger.TickerEvent) me
 	return queryMap
 }
 
-// TODO serviceEntry
 func (r *SmartLimiterReconciler) handleEvent(loc types.NamespacedName) metric.QueryMap {
 	queryMap := make(map[string][]metric.Handler, 0)
 	v, ok := r.interest.Get(FQN(loc.Namespace, loc.Name))
 	if !ok {
-		log.Infof("%s not in interest map", loc)
+		log.Warnf("%s not in interest map", loc)
 		return queryMap
 	}
-	// no need to query service information in outbound and gateway scenario
-	host := v.(string)
-	if host == model.MockHost {
-		meta := &StaticMeta{Name: loc.Name, Namespace: loc.Namespace}
-		metaInfo := meta.String()
-		queryMap[metaInfo] = []metric.Handler{}
+
+	meta, ok := v.(SmartLimiterMeta)
+	if !ok {
+		log.Error("covert to SmartLimiterMeta err")
+		return queryMap
+	}
+
+	// if it is mesh outbound or gateway, queryMap just contains namespace/name
+	if meta.outbound {
+		sm := &StaticMeta{Name: loc.Name, Namespace: loc.Namespace}
+		queryMap[sm.String()] = []metric.Handler{}
 		return queryMap
 	}
 
 	// handle loc which is in interest map in inbound scenario
 	if !r.cfg.GetDisableAdaptive() {
-		return r.handlePrometheusEvent(host, loc)
+		return r.handlePrometheusEvent(meta, loc)
 	} else {
-		return r.handleLocalEvent(host, loc)
+		return r.handleLocalEvent(meta, loc)
 	}
 	return nil
 }
 
-func (r *SmartLimiterReconciler) handleLocalEvent(host string, loc types.NamespacedName) metric.QueryMap {
+func (r *SmartLimiterReconciler) handleLocalEvent(LimiterMeta SmartLimiterMeta, loc types.NamespacedName) metric.QueryMap {
 	queryMap := make(map[string][]metric.Handler, 0)
 
-	// if it does not end with .svc.cluster.local, it means that se host is specified
-	if !strings.HasSuffix(host, util.WellknownK8sSuffix) {
+	// workloadSelector is specified
+	if len(LimiterMeta.workloadSelector) > 0 {
+		// here, there is such a semantics
+		// if it is under istioNs, it will match all the pods in cluster
+		// other it will only match the real ns in cluster
+		ns := loc.Namespace
+		if loc.Namespace == r.env.Config.Global.IstioNamespace {
+			ns = ""
+		}
+		pods, err := queryPodsWithWorkloadSelector(r.env.K8SClient, LimiterMeta.workloadSelector, ns)
+		if err != nil {
+			log.Errorf("get err in queryServicePodsWithWorkloadSelector, %+v", err.Error())
+			return nil
+		}
+
+		subsetInfo := make(map[string]int)
+		subsetInfo[util.WellknownBaseSet] = len(pods)
+
+		meta := generateMeta(subsetInfo, loc)
+		metaInfo := meta.String()
+		log.Debugf("get workloadSelector meta info %s", metaInfo)
+		if metaInfo == "" {
+			return nil
+		}
+		queryMap[metaInfo] = []metric.Handler{}
+		return queryMap
```

Comment:
we can handle different cases in separated functions

Reply:
done